pkg/tags/tags_test.go

```diff
@@ -52,3 +59,81 @@ func TestAll(t *testing.T) {
 		t.Fatalf("expected length to be 3 got %d", len(all))
 	}
 }
+
+func TestPersistence(t *testing.T) {
+	mockStatestore := statestore.NewStateStore()
+	logger := logging.New(ioutil.Discard, 0)
+	ts := NewTags(mockStatestore, logger)
+	ta, err := ts.Create("one", 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+	ta.Total = 10
+	ta.Seen = 2
+	ta.Split = 10
+	ta.Stored = 8
+	_, err = ta.DoneSplit(swarm.ZeroAddress)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// simulate node closing down and booting up
+	err = ts.Close()
+	if err != nil {
+		t.Fatal(err)
+	}
+	ts = NewTags(mockStatestore, logger)
+
+	// Get the tag after the node bootup
+	rcvd1, err := ts.Get(ta.Uid)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// check if the values ae intact after the bootup
+	if ta.Uid != rcvd1.Uid {
+		t.Fatalf("invalid uid: expected %d got %d", ta.Uid, rcvd1.Uid)
+	}
+	if ta.Total != rcvd1.Total {
+		t.Fatalf("invalid total: expected %d got %d", ta.Total, rcvd1.Total)
+	}
+
+	// See if tag is saved after syncing is over
```

Comment:
Since updating the tag on every inc() is expensive, @janos and me had a discussion before implementing this and decided that we will store in the following cases
1) When the splitting is over 
2) When the syncing is over
3) When the system does a shutdown

This is so that we can reload the tags when sync is resumed after the bootup.